src/Microsoft.Health.Fhir.Tests.Common/Mock.cs

```diff
@@ -114,12 +115,15 @@ public static object TypeWithArguments(Type type, params object[] argOverrides)
                     }
                     else
                     {
-                        arguments.Add(Substitute.For(new[] { parameter.ParameterType }, null));
+                        object item = parameter.ParameterType.IsInterface ?
+                            Substitute.For(new[] { parameter.ParameterType }, null) :
+                            SubstitutionContext.Current.SubstituteFactory.CreatePartial(new[] { parameter.ParameterType }, null);
+                        arguments.Add(item);
                     }
                 }
             }
 
-            return Substitute.For(new[] { type }, arguments.ToArray());
+            return SubstitutionContext.Current.SubstituteFactory.CreatePartial(new[] { type }, arguments.ToArray());
```

Comment:
Using this to enable behavior in `IFhirElementToSearchValueTypeConverter`s above